src/sentry_value.c

```diff
@@ -525,12 +547,8 @@ sentry__value_clone(sentry_value_t value)
         const obj_t *obj = thing->payload;
         sentry_value_t rv = sentry__value_new_object_with_size(obj->len);
         for (size_t i = 0; i < obj->len; i++) {
-            char *key = sentry__string_clone(obj->pairs[i].k);
```

Comment:
this is just lol, why? Also, another piece of code that is apparently never run, or at least not in leak-checked code?

Reply:
Potentially we didn't use `sentry_value_set_by_key` initially. To be fair, it would probably be faster to use something lower level, given that we know the size up-front and can skip a lot of checks. But please double-check why this never came up in any tests, we should at least have unit test coverage for this (preferred over integration tests, actually).

Otherwise, please add a comment that `sentry_value_set_by_key` clones the key internally.